### PR TITLE
Us 05 fatigue troupe

### PR DESCRIPTION
**Pull request by FranciscoKevin**
Résolution du bug phpstan

### CODE REVIEW

**Review by maxime-auneau — DISMISSED** [review]

File: src/Model/Troop.php [hunk 1]
```diff
@@ -67,4 +73,13 @@ public function getName(): string
     {
         return $this->name;
     }
+
+    public function setTiredness(): void
+    {
+        $this->tiredness = self::TIREDNESS_MAX;
+    }
```
Comment (RomainClair) [inline src/Model/Troop.php]: The name of the method should be resetTiredness()

File: src/Model/TroopManager.php [hunk 2]
```diff
@@ -32,12 +32,15 @@ public function __construct()
 
     public function insert(Troop $troop)
     {
+
         // prepared request
-        $insert = $this->pdo->prepare("INSERT INTO " . self::TABLE . " (name, strength) VALUES (:name, :strength)");
+        $insert = $this->pdo->prepare("INSERT INTO " . self::TABLE . " (name, strength, tiredness)
```
Comment (RomainClair) [inline src/Model/TroopManager.php:37]: there's no tiredness field in the database schema provided by the defendthecastle.sql file

**Review by RomainClair — CHANGES_REQUESTED** [review]

**Review by maxime-auneau — DISMISSED** [review]
ok

File: src/Model/Troop.php (hunk 1 above)
Comment (thocle86) [reply src/Model/Troop.php]: Modification ok

File: src/Model/TroopManager.php (hunk 2 above)
Comment (thocle86) [reply src/Model/TroopManager.php:37]: Modification ok

**Review by thocle86 — DISMISSED** [review]
The setTiredness function is passed to resetTiredness + the Tiredness column has been added to the table => so ok for me

**Review by thocle86 — DISMISSED** [review]
Resolution conflicts ok for me

**Review by maxime-auneau — DISMISSED** [review]
ok

File: defend_the_castle.sql [hunk 3]
```diff
@@ -1,76 +1,77 @@
...
+) ENGINE=InnoDB AUTO_INCREMENT=4 DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci;
+/*!40101 SET character_set_client = @saved_cs_client */;
+
+--
+-- Dumping data for table `troop`
+--
+
+LOCK TABLES `troop` WRITE;
+/*!40000 ALTER TABLE `troop` DISABLE KEYS */;
+INSERT INTO `troop` VALUES (1,'Archer',72),(2,'Horseman',52),(3,'Lancer',38);
```
Comment (RomainClair) [inline defend_the_castle.sql]: This line generate : ERROR 1136 (21S01) at line 64: Column count doesn't match value count at row 1

**Review by RomainClair — CHANGES_REQUESTED** [review]
An error in the sql dump

**Review by maxime-auneau — DISMISSED** [review]
ok

**Review by AshBll — DISMISSED** [review]

**Review by maxime-auneau — DISMISSED** [review]
ok

**Review by RomainClair — CHANGES_REQUESTED** [review]
SQL dump still have syntax errors

**Pull request closed without merging**